src/OnChainSurplus.py

```diff
@@ -0,0 +1,113 @@
+gpv2Abi = json.loads(abiJson)
+#connecting to ethereum node
+w3 = Web3(Web3.HTTPProvider(f"https://mainnet.infura.io/v3/{INFURA_KEY}"))
+#creating an instance of GPv2 contract 
+contractInstance = w3.eth.contract(address=address, abi=gpv2Abi)
+
+class DecodedSettlement:
+    def __init__(self, tokens: List[str], clearing_prices: List[int], trades: List[Tuple[int, int, str, int, int, int, bytes, int, int, int, bytes]], interactions: List[List[Tuple[str, int, bytes]]]):
+        self.tokens = tokens
+        self.clearing_prices = clearing_prices
+        self.trades = trades
+        self.interactions = interactions
+
+    @classmethod
+    def new(cls, contractInstance, transaction):
+        # Decode the function input
```

Comment:
I think there is no need for this function to receive the whole transaction, while only using the `input` field. Can we receive the `input` as the parameter directly?